restapi/src/main/java/io/stargate/web/docsapi/service/query/DocumentSearchService.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.web.docsapi.service.query;
+
+import com.bpodgursky.jbool_expressions.Expression;
+import com.bpodgursky.jbool_expressions.Literal;
+import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.core.Single;
+import io.stargate.db.datastore.DataStore;
+import io.stargate.db.query.BoundQuery;
+import io.stargate.db.query.Query;
+import io.stargate.db.query.builder.BuiltQuery;
+import io.stargate.web.docsapi.dao.Paginator;
+import io.stargate.web.docsapi.service.DocsApiConfiguration;
+import io.stargate.web.docsapi.service.ExecutionContext;
+import io.stargate.web.docsapi.service.QueryExecutor;
+import io.stargate.web.docsapi.service.RawDocument;
+import io.stargate.web.docsapi.service.query.search.db.impl.FullSearchQueryBuilder;
+import io.stargate.web.docsapi.service.query.search.db.impl.PopulateSearchQueryBuilder;
+import io.stargate.web.docsapi.service.query.search.resolver.BaseResolver;
+import io.stargate.web.docsapi.service.query.search.resolver.DocumentsResolver;
+import io.stargate.web.rx.RxUtils;
+import java.util.Collection;
+import javax.inject.Inject;
+import org.apache.commons.lang3.tuple.Pair;
+
+public class DocumentSearchService {
+
+  @Inject private DocsApiConfiguration configuration;
+
+  public Flowable<RawDocument> searchDocuments(
+      QueryExecutor queryExecutor,
+      String keyspace,
+      String collection,
+      Expression<FilterExpression> expression,
+      Collection<String> fields,
+      Paginator paginator,
+      ExecutionContext context) {
+
+    // if we have true immediately, means we can only do full search
+    if (Literal.EXPR_TYPE.equals(expression.getExprType())) {
```

Comment:
Should we test for `true` here in addition to expression type?.. just for the same of correctness. If it's literal `false` we have to return nothing, right?

Reply:
Done.